test/common.bash

```diff
@@ -326,18 +334,17 @@ ch_ttar=${ch_tardir}/chtest.tar.gz
 # shellcheck disable=SC2034
 ch_timg=${ch_imgdir}/chtest
 
-# MPICH requires different handling from OpenMPI. Set a variable to enable
-# some kludges.
 if [[ $ch_tag = *'-mpich' ]]; then
     ch_mpi=mpich
-    # First kludge. MPICH's internal launcher is called "Hydra". If Hydra sees
-    # Slurm environment variables, it tries to launch even local ranks with
-    # "srun". This of course fails within the container. You can't turn it off
-    # by building with --without-slurm like OpenMPI, so we fall back to this
-    # environment variable at run time.
-    export HYDRA_LAUNCHER=fork
+    # As of MPICH 4.0.2, using SLURM as the MPICH process manager requires two
+    # configure options that disable the compilation of mpiexec. This may not
+    # always be the case.
+    # shellcheck disable=SC2034
+    ch_mpi_exe=mpiexec
```

Comment:
Is there some other way to do within-container MPI start then?

Reply:
Not that I am aware of. Making MPICH use SLURM as the pm disables the creation of `mpiexec`.